### DISCUSSION TITLE
Arch Linux container: Expected commit object, not File

### DISCUSSION THREAD

**Discussion by M1cha** [bootc-dev/bootc#110] [Q&A]
> However, bootc itself is not tied to Fedora derivatives; and the plan is to extend the set of images.

Challenge accepted. I guess I'm porting arch now :smile: 

So I started building a simple Containerfile that would end up in an image usable by bootc. I made some progress but I need some help if you have the time.

Here's my current code:
https://github.com/M1cha/bootc-archlinux/blob/a109d236c17664786025483818d1fdbf74002262/Containerfile

As the title of this discussion hints at, I can't install the container to a disk:
```bash
# podman run --privileged --pid=host --net=none --security-opt label=type:unconfined_t bootc-arch bootc install --target-no-signature-verification /dev/sdb
Mounting devtmpfs
Initializing partitions
Creating filesystem
mke2fs 1.47.0 (5-Feb-2023)
Creating filesystem
Mounting /run/bootc/mounts/rootfs
Mounting /run/bootc/mounts/rootfs/boot
Creating ESP filesystem
Mounting /run/bootc/mounts/rootfs/boot/efi
Initializing ostree layout
Initializing sysroot
ostree/deploy/default initialized as OSTree root
Creating initial deployment
ERROR Creating ostree deployment: Performing deployment: Importing: Unencapsulating base: Importing commit: Expected commit object, not File
```

I assume that it's an issue with how I create the commit. I couldn't yet find out how exactly `rpm-ostree compose` does it, so do you maybe know that off the top of your head? It's suspicious that `ghcr.io/cgwalters/c9s-oscore` uses a repo in mode `bare-split-xattrs`. If I use that, the commit fails with `error: Writing content object: Not allowed due to repo mode`. I don't really have to use that mode though since archlinux doesn't support selinux anyway.

**Comment by savely-krasovsky** [comment]
From my understanding you don't need ostree inside image. This should be optional. Am I not right?
